Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -464,11 +477,21 @@ void TextureCacheBase::BindTextures()
 
 void TextureCacheBase::UnbindTextures()
 {
-  bound_textures.fill(nullptr);
+  for (size_t i = 0; i < bound_textures.size(); ++i)
```

Comment:
Hmmmm...

This function is actually called from two places... One is `TextureCacheBase::Invalidate()`, where you do indeed want to wipe all the textures, even if they are valid.

The other place is `VertexManagerBase::Flush()`. Arguably you don't need to call it from there at all... as it's going to try and call `TextureCacheBase::Load()` on every single texture anyway.

The one complication is `Flush()` will only call `Load()` on the fist `usedtextures` slots. Instead of Calling UnbindTextures, you might want to pass that variable into `BindTextures()` instead.